### DISCUSSION TITLE
2.16.0 Enum serialization regression ?

### DISCUSSION THREAD

**Discussion by gjardillier** [FasterXML/jackson-databind#4563] [Q&A] [answered]
Hi,
I am using jackson with spring to serialize a java enumeration which looks like this :

```
@JsonFormat(shape = JsonFormat.Shape.OBJECT)
public enum Level {
  LEVEL1("level1"),
  LEVEL2("level2"),
  LEVEL3("level3", Level.LEVEL1);

  String label;
  Level sublevel;

.......
}
```

Before updating to 2.16.0, I was getting this when serializing : 

```
[
  {
    "label": "level1"
  },
  {
    "label": "level2"
  },
  {
    "label": "level3",
    "sublevel": {
        "label": "level1"
    }
  }
]
```

Since 2.16.0, I'm getting :
```
[
  {
    "label": "level1"
  },
  {
    "label": "level2"
  },
  {
    "label": "level3"
  }
]
```
Is this the expected behaviour ?
Is there a different way to achieve the previous result with parameters or annotations ?

**Comment by JooHyukKim** [comment]
Different way of Enum handling was  applied since 2.16.
But should have not affected such behavior. 🤔

Could u share with us the rest of Level implementation and also Mapper configuration?

**Comment by cowtowncoder** [comment, marked as answer]
Sounds like this should be an issue, instead of discussion -- I will convert.

And for issue, we would need full reproduction (much of what is needed is already present, I think).

**Reply by JooHyukKim** [reply]
Mentioned issue is... at https://github.com/FasterXML/jackson-databind/issues/4564

**Reply by gjardillier** [reply]
Thank you both.
I replied in the issue.
